### ISSUE TITLE
Not able to set custom endpoint_url

### ISSUE THREAD

**Issue by Glutexo** [kislyuk/watchtower#92]
As far as I know it is not possible to set the endpoint URL in any way. Because of that is not possible to connect to a custom/local AWS instance, e.g. LocalStack.

**Comment by tblake84** [comment]
This is a much needed feature.

**Comment by tblake84** [comment]
I worked out how to fix this:

In python3.6/site-packages/watchtower/__init__.py, the __init__ function should read the following:
```
    def __init__(self, endpoint_url=None, log_group=__name__, stream_name=None, use_queues=True, send_interval=60,
                 max_batch_size=1024 * 1024, max_batch_count=10000, boto3_session=None,
                 boto3_profile_name=None, create_log_group=True, log_group_retention_days=None,
                 create_log_stream=True, json_serialize_default=None, *args, **kwargs):
        handler_base_class.__init__(self, *args, **kwargs)
        self.log_group = log_group
        self.stream_name = stream_name
        self.use_queues = use_queues
        self.send_interval = send_interval
        self.json_serialize_default = json_serialize_default or _json_serialize_default
        self.max_batch_size = max_batch_size
        self.max_batch_count = max_batch_count
        self.queues, self.sequence_tokens = {}, {}
        self.threads = []
        self.creating_log_stream, self.shutting_down = False, False
        self.create_log_stream = create_log_stream
        self.log_group_retention_days = log_group_retention_days

        # Creating session should be the final call in __init__, after all instance attributes are set.
        # This ensures that failing to create the session will not result in any missing attribtues.
        if endpoint_url is None:
            self.cwl_client = self._get_session(boto3_session, boto3_profile_name).client("logs")
        else:
            self.cwl_client = self._get_session(boto3_session, boto3_profile_name).client("logs", endpoint_url=endpoint_url)
        if create_log_group:
            _idempotent_create(self.cwl_client.create_log_group, logGroupName=self.log_group)

        if log_group_retention_days:
            _idempotent_create(
                self.cwl_client.put_retention_policy,
                logGroupName=self.log_group,
                retentionInDays=self.log_group_retention_days
            )
```

Then you can specify endpoint_url in your log handler:
```
watchtower.CloudWatchLogHandler(log_group=loggroup, stream_name='main', endpoint_url='https://' + config.logendpoint)
```

**Comment by Glutexo** [comment]
I patched it exactly the same way. Just didn’t bother to make a pull request. 😞 Shame!

**Comment by pzmosquito** [comment]
here's the PR https://github.com/kislyuk/watchtower/pull/114

**Comment by kislyuk** [comment]
The custom endpoint URL can be set by passing the `boto3_session` keyword argument.

**Comment by kislyuk** [comment]
Never mind, `endpoint_url` must be passed to the client constructor, not the session constructor.

**Comment by kislyuk** [comment]
This is addressed by merging #114.

**Comment by dparker2** [comment]
When do you expect to release this addition?
